lib/index.js

```diff
@@ -1,41 +1,33 @@
 'use strict';
 
+const Promise = require('bluebird');
 const Fs = require('fs');
-const Async = require('async');
 const Path = require('path');
 
 const testDefault = /\.(js)$/;
+const internals = {};
 
-exports.register = (server, baseOptions, next) => {
+internals.readDir = Promise.promisify(Fs.readdir, { context: Fs });
```

Comment:
Did you know that node has this built in? See https://nodejs.org/dist/latest-v8.x/docs/api/util.html#util_util_promisify_original
Please use that one, no need for more deps ^^